ui/components-react/src/components-react/filter-builder/FilterBuilderState.ts

```diff
@@ -169,3 +170,48 @@ function findRule(rootGroup: PropertyFilterBuilderRuleGroup, path: string[]): Pr
 
   return currentItem;
 }
+
+function getGroupRuleItem(filter: PropertyFilter, parentId: string): PropertyFilterBuilderRuleGroupItem {
+  const id = Guid.createValue();
+  if (isPropertyFilterRuleGroup(filter))
+    return {
+      id,
+      groupId: parentId,
+      operator: filter.operator,
+      items: filter.rules.map((rule) => getGroupRuleItem(rule, id)),
+    };
+  return getSingleRuleItem(filter, id);
+}
+
+function getSingleRuleItem(filter: PropertyFilterRule, parentId: string) {
+  return {
+    id: Guid.createValue(),
+    groupId: parentId,
+    property: filter.property,
+    operator: filter.operator,
+    value: filter.value,
+  };
+}
+
+/** @alpha */
+export function convertFilterToState(filter?: PropertyFilter): PropertyFilterBuilderState | undefined {
```

Comment:
I guess `filter` can be made required and check for `undefined` can be made before calling `convertFilterToState`.

Also, I think it would be great if we could avoid exporting this function.